src/aggregate_verify.cc

```diff
@@ -0,0 +1,226 @@
+        if (err != blst_ts::BLST_TS_ERROR::SUCCESS) {
+            return err;
+        }
+    }
+
+    return blst_ts::BLST_TS_ERROR::SUCCESS;
+}
+
+/**
+ * @param[out]  {boolean} - Result of the verification
+ * @param[out]  {std::string} - Error message for invalid aggregate
+ * @param[in]   {BlstTsAddon *} - Addon module
+ * @param[in]   {blst::Paring *} - Pointer to paring for the verification
+ * @param[in]   {blst_ts::P2AffineGroup} - P2 point for verification
+ * @param[in]   {std::vector<AggregateVerifySet>} - Sets to be added to pairing
+ */
+blst_ts::BLST_TS_ERROR aggregate_verify(
+    bool &result,
+    std::string &error_msg,
+    BlstTsAddon *module,
+    std::unique_ptr<blst::Pairing> &ctx,
+    blst_ts::P2AffineGroup &sig_point,
+    std::vector<AggregateVerifySet> &sets) {
+    for (uint32_t i = 0; i < sets.size(); i++) {
+        blst::BLST_ERROR err = ctx->aggregate(
+            sets[i].pk_point.raw_point,
+            sig_point.raw_point,
+            sets[i].msg,
+            sets[i].msg_len);
+        if (err != blst::BLST_ERROR::BLST_SUCCESS) {
+            error_msg = "BLST_ERROR::"s + module->GetBlstErrorString(err) +
+                        ": Invalid verification aggregate at index "s +
+                        std::to_string(i);
+            return blst_ts::BLST_TS_ERROR::HAS_NATIVE_ERROR;
+        }
+    }
+
+    ctx->commit();
+    blst::PT pt{*sig_point.raw_point};
+    result = ctx->finalverify(&pt);
+
+    return blst_ts::BLST_TS_ERROR::SUCCESS;
+}
+
+Napi::Value AggregateVerify(const Napi::CallbackInfo &info) {
+    BLST_TS_FUNCTION_PREAMBLE(info, env, module)
+    try {
+        bool result{false};
+        std::string error_msg{};
+        blst_ts::P2AffineGroup sig_point{};
+        std::vector<AggregateVerifySet> sets{};
+        blst_ts::BLST_TS_ERROR error =
+            prepare_aggregate_verify(sig_point, sets, info);
+        if (error == blst_ts::BLST_TS_ERROR::SUCCESS) {
+            std::unique_ptr<blst::Pairing> ctx{
+                new blst::Pairing(true, module->dst)};
+            error = aggregate_verify(
+                result, error_msg, module, ctx, sig_point, sets);
+        }
+        switch (error) {
+            case blst_ts::BLST_TS_ERROR::SUCCESS:
+                return scope.Escape(Napi::Boolean::New(env, result));
+            case blst_ts::BLST_TS_ERROR::JS_ERROR_THROWN:
+                return scope.Escape(env.Undefined());
+            case blst_ts::BLST_TS_ERROR::HAS_NATIVE_ERROR:
+                Napi::Error::New(env, error_msg).ThrowAsJavaScriptException();
+                return scope.Escape(env.Undefined());
+            default:
+            case blst_ts::BLST_TS_ERROR::INVALID:
+                return scope.Escape(Napi::Boolean::New(env, false));
+        }
+    } catch (...) {
+        return scope.Escape(Napi::Boolean::New(env, false));
+    }
+}
+
+class AggregateVerifyWorker : public Napi::AsyncWorker {
```

Comment:
Same comments as in `VerifyMultipleAggregateSignaturesWorker`. In general, would it make sense to have some shared code between those two classes?

Reply:
That is what the `functions-inl.h` was meant to be.  It's the shared helper functions that are used by both files